sympy/physics/mechanics/body.py

```diff
@@ -488,11 +488,9 @@ def masscenter_vel(self, body):
 
         """
 
-        if isinstance(body,  ReferenceFrame):
-            frame=body
-        elif isinstance(body, Body):
-            frame = body.frame
-        return self.masscenter.vel(frame)
+        if isinstance(body, Body):
+            body = body.frame
+        return self.masscenter.vel(body)
```

Comment:
Class `Body` doesn't need these changes.

Reply:
Noted. I thought this suffered the same problem as the other one where I made this change, so I added it here too.

Reply:
No remove changes from Body.